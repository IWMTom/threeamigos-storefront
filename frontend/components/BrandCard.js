import { Link } from "../routes";

const BrandCard = ({ brand }) => (
	<div className="pure-u-1-2">
		<Link route="brand" params={{ slug: brand.slug }}>
			<a className="card brand">
				<img src={brand.image_url} alt={brand.name} />
			</a>
		</Link>
	</div>
);

export default BrandCard;
