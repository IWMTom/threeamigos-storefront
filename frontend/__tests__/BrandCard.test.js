import BrandCard from "../components/BrandCard";
import { shallow, mount } from "enzyme";
import toJSON from "enzyme-to-json";

const mockedBrand = {
	id: "abcdefghijklmnopqrst",
	slug: "istuff-r-us",
	name: "iStuff-R-Us",
	image_url: "http://placehold.it/300x100"
};

describe("<BrandCard />", () => {
	it("renders and matches the snapshot", () => {
		const wrapper = shallow(<BrandCard brand={mockedBrand} />);

		expect(toJSON(wrapper)).toMatchSnapshot();
	});
});
