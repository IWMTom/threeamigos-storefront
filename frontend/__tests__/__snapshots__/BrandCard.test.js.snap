// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BrandCard /> renders and matches the snapshot 1`] = `
<div
  className="pure-u-1-2"
>
  <LinkRoutes
    params={
      Object {
        "slug": "istuff-r-us",
      }
    }
    route="brand"
  >
    <a
      className="card brand"
    >
      <img
        alt="iStuff-R-Us"
        src="http://placehold.it/300x100"
      />
    </a>
  </LinkRoutes>
</div>
`;
